methods.rb

```diff
@@ -0,0 +1,146 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    until i == arr.length
+      yield(arr[i])
+      i += 1
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    until i == arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    arr
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each do |i|
+      arr << [i] if yield(i)
+    end
+    arr
+  end
```

Comment:
#### my_select 

- [ ] returns an array containing all elements of enum for which the given block returns a true value

Given

  block = proc { |num| num < (0 + 9) / 2 } 

Expected
   
>  [2,4,7,11].my_select(&block) should be equal to [2,4,7,11].select(&block)   => [2] not [[2]]